### PR TITLE
Check 140 - Wire up create new org form on admin end to call API

### PR DESCRIPTION
**Pull request by digitalsoba**
# Description
Sign up form now hits `api/register_admin` backend and creates new organization and admin user.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
- Load up the docker environment 
- Visit [http://localhost:8080/#/signup](http://localhost:8080/#/signup)
- Open up a console log
- Fill in form and submit. What should return is a similar payload to the following
```json
{
    "organization": {
        "organization_name": "example@email.com",
        "logo": {}
    },
    "user": {
        "name": "Hulk Hogan",
        "email": "example@email.com"
    }
}
```

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by JayJayAquino — COMMENTED** [review]
:+1:

**Review by chippingcolt — APPROVED** [review]
Check checkpoint-native api directory and mutation-types.

File: resources/src/js/router/views/signup/index.vue [hunk 1]
```diff
@@ -6,68 +6,163 @@
...
+          this.imageUrl = fr.result;
+          this.imageFile = files[0];
+        });
+      } else {
+        this.imageName = "";
+        this.imageFile = "";
+        this.imageUrl = "";
       }
+    },
+    submit() {
```
Comment (404tonynotfound) [inline resources/src/js/router/views/signup/index.vue:142]: For completeness lets put this into vuex, there's a convention we use with API calls and vuex, you can follow calstatepays UI code for an example of it

File: resources/src/js/router/views/signup/index.vue [hunk 2]
```diff
@@ -6,68 +6,163 @@
...
-          lastName: { required },
-          email: { required },
-          password: { required },
-          validatePassword: { required },
-          address1: { required },
-          address2: { required },
-          city: { required },
-          country: { required },
-          zip: { required }
+    form: {
```
Comment (404tonynotfound) [inline resources/src/js/router/views/signup/index.vue:83]: I'm not sure these validations will work... your data properties should also be in a form object with this

File: resources/src/js/router/views/signup/index.vue [hunk 3]
```diff
@@ -6,68 +6,163 @@
...
+      city: null,
+      country: null,
+      zip_code: null,
+      state: null,
+      imageName: null,
+      imageUrl: null,
+      imageFile: null
+    };
+  },
+  computed: {},
```
Comment (404tonynotfound) [inline resources/src/js/router/views/signup/index.vue:118]: You can scrap this line

File: resources/src/js/router/views/signup/index.vue [hunk 4]
```diff
@@ -6,68 +6,163 @@
...
+  methods: {
+    pickFile() {
+      this.$refs.image.click();
```
Comment (404tonynotfound) [inline resources/src/js/router/views/signup/index.vue:121]: if the method is a one liner and only used once, we can just put this in the @click="" area

File: resources/src/js/router/views/signup/index.vue [hunk 5]
```diff
@@ -6,68 +6,163 @@
...
+        });
+      } else {
+        this.imageName = "";
+        this.imageFile = "";
+        this.imageUrl = "";
       }
+    },
+    submit() {
+      axios
+        .post("/api/register_admin", {
```
Comment (404tonynotfound) [inline resources/src/js/router/views/signup/index.vue:144]: So if you move all your data properties into a form object, you can just pass the form object in, like 
`axios.post('/api/register_admin', this.form);`

**Review by 404tonynotfound — CHANGES_REQUESTED** [review]

**Pull request closed without merging**